src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -21,35 +28,202 @@ const ActorInit En_Js_InitVars = {
+    s16 temp;
+    s32 temp_2;
+
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        return 1;
+    } else {
+        this->actor.textId = arg2;
+        temp = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+
+        if (temp >= 0) {
+            temp_2 = temp;
+        } else {
+            temp_2 = -temp;
+        }
+
+        if (temp_2 < 0x1801 && this->actor.xzDistFromLink < 100.0f) {
+            this->unk_284 |= 1;
+            func_8002F2CC(&this->actor, globalCtx, 100.0f);
+        }
+        return 0;
+    }
+}
+
+void func_80A89008(EnJs* this) {
+    En_Js_SetupAction(this, func_80A89304);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600045C, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_0600045C.genericHeader),
+                         2, -4.0f);
+}
+
+void func_80A89078(EnJs* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx) != 0) {
+        func_80A89008(this);
+        this->actor.flags &= ~0x10000;
+    }
+}
+
+void func_80A890C0(EnJs* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        En_Js_SetupAction(this, func_80A89078);
+    } else {
+        func_8002F2CC(&this->actor, globalCtx, 1000.0f);
+    }
+}
+
+void func_80A8910C(EnJs* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx) != 0) {
+        this->actor.textId = 0x6078U;
+        En_Js_SetupAction(this, func_80A890C0);
+        this->actor.flags |= 0x10000;
+    }
+}
+
+void func_80A89160(EnJs* this, GlobalContext* globalCtx) {
+    if (func_8002F410(&this->actor, globalCtx) != 0) {
+        this->actor.attachedA = NULL;
+        En_Js_SetupAction(this, func_80A8910C);
+    } else {
+        func_8002F434(&this->actor, globalCtx, GI_BOMBCHUS_10, 10000.0f, 50.0f);
+    }
+}
+
+void func_80A891C4(EnJs* this, GlobalContext* globalCtx) {
+    u8 temp_v0;
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && (func_80106BC8(globalCtx) != 0)) {
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0: // yes
+                if (gSaveContext.rupees < 200) {
+                    // yes, but not enough rupees
+                    func_8010B720(globalCtx, 0x6075);
+                    func_80A89008(this);
+                } else {
+                    Rupees_ChangeBy(-200);
+                    En_Js_SetupAction(this, func_80A89160);
+                }
+                break;
+            case 1: // no
+                func_8010B720(globalCtx, 0x6074);
+                func_80A89008(this);
+        }
+    }
+}
+
+void func_80A89294(EnJs* this) {
+    En_Js_SetupAction(this, func_80A891C4);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600018C, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_0600018C.genericHeader),
+                         2, -4.0f);
+}
+
+void func_80A89304(EnJs* this, GlobalContext* globalCtx) {
+    // 0x6077 - text for initial offer to buy something for 200 rupees
```

Comment:
we're trying to stay away from comments describing text. eventually the whole text table will be in here and youll be able to look them up easily